src/context/common.ml

```diff
@@ -588,6 +592,11 @@ let init_platform com pf =
 		define com Define.Sys
 	end else
 		com.package_rules <- PMap.add "sys" Forbidden com.package_rules;
+	if com.config.pf_asys then begin
+		raw_define_value com.defines "target.asys" "true";
+		define com Define.Asys
```

Comment:
Why have both `target.asys` and `asys`? I thought the defines not prefixed by `target.` are just for backwards compatibility.

Reply:
If we have `sys` we should also have `asys`, otherwise this is just one giant trap.